pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicPoliciesWithBrokerRestartTest.java

```diff
@@ -0,0 +1,100 @@
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Test(groups = "broker-admin")
+public class TopicPoliciesWithBrokerRestartTest extends MockedPulsarServiceBaseTest {
+
+    @Override
+    @BeforeClass(alwaysRun = true)
+    protected void setup() throws Exception {
+        super.internalSetup();
+        setupDefaultTenantAndNamespace();
+    }
+
+    @Override
+    @AfterClass(alwaysRun = true)
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+    }
+
+
+    @Test
+    public void testRetentionWithBrokerRestart() throws Exception {
+        final int messages = 1_000;
+        // (1) Init topic
+        admin.namespaces().createNamespace("public/retention");
+        final String topicName = "persistent://public/retention/retention_with_broker_restart";
+        admin.topics().createNonPartitionedTopic(topicName);
+        for (int i = 0; i < 500; i++) {
+            final String shadowTopicNames = topicName + "_" + i;
+            admin.topics().createNonPartitionedTopic(shadowTopicNames);
+        }
+        // (2) Set retention
+        final RetentionPolicies retentionPolicies = new RetentionPolicies(20, 20);
+        for (int i = 0; i < 500; i++) {
+            final String shadowTopicNames = topicName + "_" + i;
+            admin.topicPolicies().setRetention(shadowTopicNames, retentionPolicies);
+        }
+        admin.topicPolicies().setRetention(topicName, retentionPolicies);
+        // (3) Send messages
+        @Cleanup
+        final Producer<byte[]> publisher = pulsarClient.newProducer()
+                .topic(topicName)
+                .create();
+        for (int i = 0; i < messages; i++) {
+            publisher.send((i + "").getBytes(StandardCharsets.UTF_8));
+        }
+        // (4) Check configuration
+        Awaitility.await().untilAsserted(() -> {
+            final PersistentTopic persistentTopic1 = (PersistentTopic)
+                    pulsar.getBrokerService().getTopic(topicName, true).join().get();
+            final ManagedLedgerImpl managedLedger1 = (ManagedLedgerImpl) persistentTopic1.getManagedLedger();
+            Assert.assertEquals(managedLedger1.getConfig().getRetentionSizeInMB(), 20);
+            Assert.assertEquals(managedLedger1.getConfig().getRetentionTimeMillis(),
+                    TimeUnit.MINUTES.toMillis(20));
+        });
+        // (5) Restart broker
+        restartBroker();
+        // (6) Check configuration again
+        admin.lookups().lookupTopic(topicName);
+        final PersistentTopic persistentTopic2 = (PersistentTopic)
+                pulsar.getBrokerService().getTopic(topicName, true).join().get();
+        final ManagedLedgerImpl managedLedger2 = (ManagedLedgerImpl) persistentTopic2.getManagedLedger();
+        Assert.assertEquals(managedLedger2.getConfig().getRetentionSizeInMB(), 20);
+        Assert.assertEquals(managedLedger2.getConfig().getRetentionTimeMillis(),
+                TimeUnit.MINUTES.toMillis(20));
```

Comment:
You have created 500 topics, why not check all of them?

Reply:
Applied this comment. :)